### DISCUSSION TITLE
Want to change the Disabled control's font color

### DISCUSSION THREAD

**Discussion by amitha-achar** [MahApps/MahApps.Metro#4480] [Q&A]
Hi, I have an application where , when we run long running process, the UI will b disabled. But it was observed that the textbox content's font color was very light and not readable. So i am trying to fix it. In App.xaml, i tried to change the style and added a trigger which will change foreground color to Black if the enabled= false. But this is changing the original style of the textbox. It shows with 4 borders for the textbox. Is it possible t o just change the DisabledForegroundBrush to something darker?or can i chnage the opacity of the disabled control to something brighter.I am on this from 4 days now and i am on the verge of giving up.

<Style BasedOn="{StaticResource MetroTextBox}" TargetType="{x:Type TextBox}">
    <Style.Triggers>
    <Trigger Property="IsEnabled" Value="False">
    <Setter Property="Foreground" Value="Black" />
    </Trigger>
    </Style.Triggers>
    </Style>

**Comment by timunie** [comment]
best would be to check original style. I believe you need to change these lines: 

https://github.com/MahApps/MahApps.Metro/blob/509c9d7e559025754854e4472db3b3adaaf9ef38/src/MahApps.Metro/Styles/Controls.TextBox.xaml#L185-L187 

> [!NOTE]
> `MetroTextBox` is outdated iirc any has a new key since 2.0. See it in the same file ^

**Reply by amitha-achar** [reply]
Thanks, I have already tried this. the moment i touch the style of metrotextbox with triggers, the Look of TextBoxes changes.The textbox starts looking like the classic textbox. But without the style, it looks different. Adding screenshot of textboxes with and without the styles
![image](https://github.com/MahApps/MahApps.Metro/assets/122306230/8e64f893-754c-476e-8b7a-9baa249971e3)

**Comment by punker76** [comment]
Please verify that you inherit from the base style, what you already showed. Maybe you have another bug in your style. You can put this here in, so maybe we can help you. But I will not help in bugs of this version of MahApps itself!
